modules/nextflow/src/main/groovy/nextflow/container/ContainerBuilder.groovy

```diff
@@ -65,8 +67,13 @@ abstract class ContainerBuilder<V extends ContainerBuilder> {
         return (V)this
     }
 
-    V setCpus( String value ) {
-        this.cpus = value
+    V setCpus(String value) {
+        this.cpus = String.format("%.1f", Float.parseFloat(value))
+        return (V)this
+    }
```

Comment:
The type should change accordingly to `Float` and formatting moved into the CLI option rendering